fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/OptOlapPartitionPruner.java

```diff
@@ -408,6 +409,8 @@ private static boolean isNeedFurtherPrune(List<Long> candidatePartitions, Logica
                 return (FunctionSet.DATE_TRUNC.equalsIgnoreCase(functionName)
                         || FunctionSet.TIME_SLICE.equalsIgnoreCase(functionName))
                         && !exprPartitionInfo.getIdToRange(true).containsKey(candidatePartitions.get(0));
+            } else if (partitionExpr.size() == 1 && partitionExpr.get(0) instanceof SlotRef) {
+                return !exprPartitionInfo.getIdToRange(true).containsKey(candidatePartitions.get(0));
```

Comment:
Can we handle all partition exprs? eg: date_trunc/str2date/cast?

Reply:
This pr is to make partition prune behavior consisitent to base table and mv, eg: `date_format` can be used to prune partition to base table, but not to mv.

Currently, it's consistent for base table and mv on these 3 functions: 
1. `date_trunc` can apply partition prune to base table and mv
2. `str2date` can't apply partition prune to base table and mv
3. `cast` can't apply partition prune to base table and mv

We can add partition prune support to `str2date/cast ` in another pr.

Reply:
I got your meaning. So this pr only to optimize partition pruning with `slot ref` partitions.